### PR TITLE
Process: Submitting an Integration Listing to the Integrations Marketplace

### PR DESCRIPTION
**Pull request by aaronrothschild**
On Plugins page: added info about Marketplace and integrations directory so developers understand how their plugins can be distributed

New Page for Submissions to Integrations Directory.

When we start accepting community plugins to the Plugin Marketplace, they will follow this same (or possibly modified) process.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-19634

### CODE REVIEW

File: site/content/contribute/server/plugins.md [hunk 1]
```diff
@@ -23,12 +23,22 @@ The server binary is a compiled (generally Go) program that extends the [Matterm
...
 
 Check out the [`plugin` package](https://github.com/mattermost/mattermost-server/tree/master/plugin) and the [plugin_* files in the `app` package](https://github.com/mattermost/mattermost-server/tree/master/app) for the code, and [mattermost-plugin-demo](https://github.com/mattermost/mattermost-plugin-demo) for an example plugin.
 
+## Plugin Marketplace
+
+The Plugin Marketplace is designed as a catalog of plugins from Mattermost and the community at large. Mattermost runs a server with a curated catalog of plugins which can then be displayed directly within the Mattermost interface ( Main Menu > Plugin Marketplace ).  You can learn more about the [Marketplace here](/administration/plugins.html#plugin-marketplace), including how to run your own Marketplace server.
```
Comment (hanzei) [inline site/content/contribute/server/plugins.md]: The link doesn't work on the CI instance http://mattermost-developer-documentation.s3-website-us-east-1.amazonaws.com/commits/33c9bf9d08a61bb6fee18561b3d27e32ee46fb3d/contribute/server/plugins/

File: site/content/contribute/server/plugins.md [hunk 2]
```diff
@@ -23,12 +23,22 @@ The server binary is a compiled (generally Go) program that extends the [Matterm
...
-
 All these different components of a plugin are compressed into a .tar.gz bundle. Installing a plugin is the process of uploading this bundle to the Mattermost server (via the UI, REST API or CLI). The server then unpacks the bundle, performs some validation and extracts it into the configured directory for storing installed plugins. Installed plugins are not yet running. To start a plugin it must be enabled (again via the UI, REST API or CLI). Once it is enabled, the server will then start the server process and prepare the web app bundle for serving to the client. Plugin settings, configuration and enabled/disabled status are managed by the Mattermost `config.json` using a [PluginSettings](https://godoc.org/github.com/mattermost/mattermost-server/model#PluginSettings) struct.
 
 Check out the [`plugin` package](https://github.com/mattermost/mattermost-server/tree/master/plugin) and the [plugin_* files in the `app` package](https://github.com/mattermost/mattermost-server/tree/master/app) for the code, and [mattermost-plugin-demo](https://github.com/mattermost/mattermost-plugin-demo) for an example plugin.
 
+## Plugin Marketplace
```
Comment (hanzei) [inline site/content/contribute/server/plugins.md]: I'm wondering if we want to have this alongside the other plugin docs. What do you think about moving it to http://mattermost-developer-documentation.s3-website-us-east-1.amazonaws.com/commits/33c9bf9d08a61bb6fee18561b3d27e32ee46fb3d/extend/plugins/? I suppose more people will see it there.

File: site/content/extend/plugins/integrations_directory_process.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+date: 2018-10-01T00:00:00-05:00
+subsection: Plugins (Beta)
+weight: 50
+---
+
+
+This document outlines the process of getting your plugin onto the [Integrations Directory](https://integrations.mattermost.com).  
+
+When submitting an Integration, keep the following in mind:
+- An Integration Directory Listing can be as simple as a blog post with instructions on how to configure an integration manually, a script, a plugin for another platform that connects with Mattermost, or a native Mattermost plugin.
```
Comment (hanzei) [inline site/content/extend/plugins/integrations_directory_process.md]: You need a line break between the lines.

File: site/content/extend/plugins/integrations_directory_process.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+subsection: Plugins (Beta)
+weight: 50
+---
+
+
+This document outlines the process of getting your plugin onto the [Integrations Directory](https://integrations.mattermost.com).  
+
+When submitting an Integration, keep the following in mind:
+- An Integration Directory Listing can be as simple as a blog post with instructions on how to configure an integration manually, a script, a plugin for another platform that connects with Mattermost, or a native Mattermost plugin.  
+- Mattermost staff will review your plugin for functionality and clarity. If the description of your plugin is not clear, we will help by submitting a Pull Request with some suggestions.
```
Comment (hanzei) [inline site/content/extend/plugins/integrations_directory_process.md]: You where talking about 
```suggestion
- Mattermost staff will review your integration for functionality and clarity. If the description of your plugin is not clear, we will help by submitting a Pull Request with some suggestions.
```

Comment (hanzei) [inline site/content/extend/plugins/integrations_directory_process.md]: Do we really want to commit our self on fixing this issues?

File: site/content/extend/plugins/integrations_directory_process.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+- If possible, please enable the "Issues" tab in Github so users can contribute and collaborate on the integration.
```
Comment (hanzei) [inline site/content/extend/plugins/integrations_directory_process.md]: Can be open the scope a bit and just ask for an issue tracker?

File: site/content/extend/plugins/integrations_directory_process.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+- Please be responsive to our questions or suggestions, we curate the integrations directory for Mattermost admins to easily find useful integrations and will work with you to make sure your listing is accurate.
+- Mattermost may not have the software available to test an integration, in which case we could do a review over a screenshare to verify functionality.
+- Your submission doesn't guarantee it will be accepted to the integrations directory.
```
Comment (hanzei) [inline site/content/extend/plugins/integrations_directory_process.md]: I'm curious why you added this. It seams discouraging to me. If my integration fulfils the checklist, why should it not get accepted?

File: site/content/extend/plugins/integrations_directory_process.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+- Your submission should contain a license file compatible with Apache 2.0. Find a list of compatible licenses here: https://apache.org/legal/resolved.html#category-a
+
+
+## Submit for Review
+
+When you are ready to have your plugin/integration start this process, please submit this form: https://spinpunch.wufoo.com/forms/mattermost-integrations-and-installers/
+
+### What will happen after submission:
+
+A Jira ticket will be created in mattermost.atlassian.net to track the progress of the listing review.  You will be sent a link to the Jira ticket within 1-2 days of submitting the form.
```
Comment (hanzei) [inline site/content/extend/plugins/integrations_directory_process.md]: How are we sending this?

File: site/content/extend/plugins/integrations_directory_process.md [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+- Ensure there is documentation to support the plugin (may include having sufficient helper text in the plugin)
+- Review any high level dependencies or security concerns
+
+#### Development: Basic Code Review
+
+This part of the process begins after the PM/UX pass.  Basic code review of an experimental plugin involves a quick review by a [core committer](/contribute/getting-started/core-committers/) to verify that the plugin does what it says it does and to provide any guidance and feedback.  We will be in contact with you if we cannot list your integration for some reason.
+
+#### Development: Second Review
+
+If changes are requested by Mattermost Development Teams, your code can be re-submitted for re-review anytime.
```
Comment (hanzei) [inline site/content/extend/plugins/integrations_directory_process.md]: What is the point of this being it's own paragraph? Why not merge it with the prior one?

File: site/content/extend/plugins/integrations_directory_process.md [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+A PM/UX pass involves a Product Manager reviewing and helping to smooth out any user experience issues with the plugin.  Generally happens within 0-2 weeks of submission.
+
+Steps involved:
+- Review or Create a one paragraph summary of the integration
+- Document the main use cases into bullet form
+- Review the primary use cases and run through them to ensure they are complete, clear and functional.
+- Ensure there is documentation to support the plugin (may include having sufficient helper text in the plugin)
+- Review any high level dependencies or security concerns
+
+#### Development: Basic Code Review
```
Comment (hanzei) [inline site/content/extend/plugins/integrations_directory_process.md]:
```suggestion
#### Basic Code Review
```

File: site/content/extend/plugins/integrations_directory_process.md [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+
+## Submit for Review
+
+When you are ready to have your plugin/integration start this process, please submit this form: https://spinpunch.wufoo.com/forms/mattermost-integrations-and-installers/
+
+### What will happen after submission:
+
+A Jira ticket will be created in mattermost.atlassian.net to track the progress of the listing review.  You will be sent a link to the Jira ticket within 1-2 days of submitting the form. 
+
+#### PM/UX Review
```
Comment (hanzei) [inline site/content/extend/plugins/integrations_directory_process.md]: By enumerating it we make clear what the steps are.

```suggestion
#### 1) PM/UX Review
```

File: site/content/extend/plugins/integrations_directory_process.md [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+This part of the process begins after the PM/UX pass.  Basic code review of an experimental plugin involves a quick review by a [core committer](/contribute/getting-started/core-committers/) to verify that the plugin does what it says it does and to provide any guidance and feedback.  We will be in contact with you if we cannot list your integration for some reason.
+
+#### Development: Second Review
+
+If changes are requested by Mattermost Development Teams, your code can be re-submitted for re-review anytime. 
+
+
+#### Listing Published
+
+Mattermost will announce the new listing via Twitter after it is published.
```
Comment (hanzei) [inline site/content/extend/plugins/integrations_directory_process.md]: Maybe we also want to send swag?

**Comment by hanzei** [comment]
@aaronrothschild You have a lot of double white spaces in your file.

**Comment by justinegeffen** [comment]
> @aaronrothschild You have a lot of double white spaces in your file.

From what I can see when I open the preview only a single space is displayed. It's always a good idea to clean them up anyway, just for consistency's sake. :)

**Comment by amyblais** [comment]
@aaronrothschild For reference, there is an existing internal process for submitting to the Integrations page: https://docs.mattermost.com/process/integrations-directory.html.

**Comment by hanzei** [comment]
@aaronrothschild Are you fine with closing this PR as you aren't working on it actively any longer?

**Pull request closed without merging**